lib/identity_cache/memoized_cache_proxy.rb

```diff
@@ -30,34 +30,47 @@ def with_memoization(&block)
     end
 
     def write(key, value)
-      memoized_key_values[key] = value if memoizing?
-      @cache_fetcher.write(key, value)
+      memoizing = memoizing?
+      ActiveSupport::Notifications.instrument('identity_cache.backend.write', memoizing: memoizing) do
+        memoized_key_values[key] = value if memoizing
+        @cache_fetcher.write(key, value)
+      end
     end
 
     def delete(key)
-      memoized_key_values.delete(key) if memoizing?
-      result = @cache_fetcher.delete(key)
-      IdentityCache.logger.debug { "[IdentityCache] delete #{ result ? 'recorded'  : 'failed'  } for #{key}" }
-      result
+      memoizing = memoizing?
+      ActiveSupport::Notifications.instrument('identity_cache.backend.delete', memoizing: memoizing) do
+        memoized_key_values.delete(key) if memoizing
+        result = @cache_fetcher.delete(key)
+        IdentityCache.logger.debug {"[IdentityCache] delete #{ result ? 'recorded' : 'failed'  } for #{key}"}
+        result
+      end
     end
 
     def fetch(key)
       used_cache_backend = true
       missed = false
-      value = if memoizing?
-        used_cache_backend = false
-        memoized_key_values.fetch(key) do
-          used_cache_backend = true
-          memoized_key_values[key] = @cache_fetcher.fetch(key) do
+      memoizing = memoizing?
+      value = ActiveSupport::Notifications.instrument('identity_cache.backend.fetch', { memoizing: memoizing }) do |payload|
```

Comment:
What I'm saying is that you **are** instrumenting the case where it is fetched an already memoized value.  The problem is that it would be fast and thus could easily skew the statistics to make it seem like a cache fetch against memcached is much faster than it actually is.